### PR TITLE
Get rid of cached db

### PR DESCRIPTION
**Pull request by jonatanklosko**
There's some weird procedure of generating a file with arrays, which imitate DB query results. Then we read this file and use the saved variables instead of queries. The queries are mostly very fast (~0.00s). For two very slow ones I found a way to make them as fast as the rest =)

### CODE REVIEW

File: WcaOnRails/db/migrate/20170503205810_add_index_on_results_country_id.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class AddIndexOnResultsCountryId < ActiveRecord::Migration[5.0]
+  def change
+    add_index :Results, :countryId
```
Comment (jonatanklosko) [inline WcaOnRails/db/migrate/20170503205810_add_index_on_results_country_id.rb]: This speeds up [this](https://github.com/thewca/worldcubeassociation.org/pull/1591/files#diff-690a237f897949397fe138a647116656R145) query.

File: webroot/results/includes/WCAClasses/WCADBConn.class.php [hunk 2]
```diff
@@ -41,109 +41,3 @@ public function getFullCompetitionInfos($id)
...
 
 }
-
-/**
-
-SOME OLD CODE HERE, FOR REFERENCE.
-
-The above function should contain all functionality in the old _database.php file, except functions below.  Need to make a class out of this or something?
-
-Should really find a better way to cache dynamic common data, rather than just including static array data as a global.  But why cache at all?  Seems like this is
```
Comment (jfly) [inline webroot/results/includes/WCAClasses/WCADBConn.class.php]: LOL this person was speaking to you =)

File: webroot/results/includes/_database.php [hunk 3]
```diff
@@ -24,7 +17,7 @@ function establishDatabaseAccess () {
   # TODO: Upgrade
   mysql_connect( $db_config['host'], $db_config['user'], $db_config['pass'] )
     or showDatabaseError( "Unable to connect to the database." );
-    
+
```
Comment (jonatanklosko) [inline webroot/results/includes/_database.php]: @jfly Oops, sorry :unamused:

File: webroot/results/includes/_database.php [hunk 4]
```diff
@@ -129,50 +122,57 @@ function dbCommand ( $command ) {
...
+    ORDER BY country.name
+  ");
 }
 
 #----------------------------------------------------------------------
 function getAllUsedContinents () {
 #----------------------------------------------------------------------
-  global $cachedUsedContinents;
-  return $cachedUsedContinents;
+  return dbQuery("SELECT * FROM Continents");
```
Comment (jonatanklosko) [inline webroot/results/includes/_database.php]: So far there's been [an expensive query](https://github.com/thewca/worldcubeassociation.org/pull/1591/files#diff-cbcbe0a5128fb29a366b2533170270aeL46) for that, but obviously all continents are *used*!

File: webroot/results/includes/WCAClasses/WCADBConn.class.php (hunk 2 above)
Comment (jonatanklosko) [inline webroot/results/includes/WCAClasses/WCADBConn.class.php]: Haha, indeed!

**Pull request merged**